### PR TITLE
compare the floatval of the divisor to 0.0

### PR DESCRIPTION
**Pull request by mgratch**
This worked exactly as expected, I did not run any unit tests though.

### CODE REVIEW

File: Visitor/Arithmetic.php [hunk 1]
```diff
@@ -185,7 +185,7 @@ public function visit(
                 if (null  === $parent ||
                     $type === $parent->getId()) {
                     $acc = function ($b) use ($a, $acc) {
-                        if (0 === $b) {
+                        if (0.0 === floatval($b)) {
```
Comment (Hywan) [inline Visitor/Arithmetic.php]: Actually, I am not sure if we can something different than a float here. I am pretty sure we always have a float, so the cast is not needed.

File: Visitor/Arithmetic.php [hunk 2]
```diff
@@ -285,7 +285,7 @@ public function getFunctions()
      * Get a function.
      *
      * @param   string  $name    Function name.
-     * @return  \Hoa\Consistency\Xcallable
+     * @return  \Hoa\Core\Consistency\Xcallable
```
Comment (Hywan) [inline Visitor/Arithmetic.php]: Nop, please, remove this. `Hoa\Core` does no longer exist (see http://blog.hoa-project.net/posts/36-the-nucleus-releases-serie.html).

File: Visitor/Arithmetic.php [hunk 3]
```diff
@@ -8,7 +8,7 @@
  *
  * New BSD License
  *
- * Copyright © 2007-2016, Hoa community. All rights reserved.
+ * Copyright © 2007-2015, Hoa community. All rights reserved.
```
Comment (Hywan) [inline Visitor/Arithmetic.php]: Please, keep the copyright ;-).

File: Visitor/Arithmetic.php [hunk 4]
```diff
@@ -44,7 +44,7 @@
  *
  * Evaluate arithmetical expressions.
  *
- * @copyright  Copyright © 2007-2016 Hoa community
+ * @copyright  Copyright © 2007-2015 Hoa community
```
Comment (Hywan) [inline Visitor/Arithmetic.php]: Please, keep the copyright ;-).

**Comment by Hywan** [comment]
@mgratch Good work, thanks! I will write the test case on my own.

**Comment by Hywan** [comment]
@mgratch ping?

**Comment by mgratch** [comment]
Pong?

**Comment by mgratch** [comment]
Did you need me to redo the PR with the changes you outlined above?  I don't have a ton of experience collaborating via GH so I apologize if I missed something.

**Comment by Hywan** [comment]
@mgratch Ok, here is what you need to do :-).
First, you need to “rebase on master”, it means your version was outdated. You need to take your commit and put them on top of the updated `master` branch. Unfortunately, you did this on the `master` branch. So first, let's rename it:
```sh
$ git checkout master # Now we are in the `master` branch
$ git branch -m master incoming # The `master` is renamed to `incoming`
```
Then, update the `master` branch:
```sh
$ git checkout -b master origin/master # Create the `master` branch from the origin remote.
$ git pull origin master:master # Pull means fetch + merge by default.
```
(assuming `origin` is the remote for this repository, you can check it by using `git remote -v`)
Then, you can “rebase” your commits:
```sh
$ git checkout incoming # Go back to incoming.
$ git rebase master # Rebase.
```
Now, you need to push the `incoming` branch. Because Github does not allow to change the reference branch of a PR, here is the trick:
```sh
$ git push origin incoming:master -f # You force to push your local `incoming` branch to the remote `master` branch.
```
This is solve most of the issue here.

Then, create a new commit to fix:
```php
if (0.0 === floatval($b))
```
to this:
```php
if (0.0 === $b)
```
Commit, and push with `git push origin incoming:master` (no need to force now, the history is clean).

You can read [our Contribution Guide](http://hoa-project.net/En/Literature/Contributor/Guide.html), it might help you. Some links are present to Git documentations or useful resources.

If you are not comfortable with this, feel free to tell me. we can do this for you :-).

**Comment by mgratch** [comment]
I will take care of this today, it might be a few hours.  Thanks for the
detailed info, I really appreciate it.

**Comment by mgratch** [comment]
So rebasing didn't fix the out of date file, likely because I probably copy & pasted the contents of the old file over the new file at some point when trying to get the original PR setup, honestly I am not sure.

But I did follow the instructions, rename my local branch, merged it with the latest version, and applied the new fix.

Let me know if this right or not.

Thank again for all your guidance.

**Comment by Hywan** [comment]
Excellent :-).
The commit message is not exactly what we expect but I can update this. Are you sure now it work?

**Comment by Hywan** [comment]
Closed by e10421c2871ce0fd7a111822b1b79f93ac04fb56.
I added the test case here: https://github.com/hoaproject/Math/commit/d7af41ed77f76a23a5cbacd3e3bb697a4b0e6d72.

**Comment by Hywan** [comment]
Thanks!

**Comment by Hywan** [comment]
Welcome on board http://hoa-project.net/En/Community.html#Contributors!

**Comment by Hywan** [comment]
Because this is a bug fix, a snapshot has been released: https://github.com/hoaproject/Math/releases/tag/1.16.01.29.

**Pull request closed without merging**